### PR TITLE
Adding {:options "-functionSyntax:4"} to standard library

### PR DESCRIPTION
**Pull request by stefan-aws**
Introducing Dafny 4 syntax, so we have less work later on.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: src/Collections/Sequences/LittleEndianNatConversions.dfy [hunk 1]
```diff
@@ -222,15 +222,15 @@ abstract module LittleEndianNatConversions {
 }
 
 /* Conversions between sequences of uint8 and uint16. */
-module Uint8_16 refines LittleEndianNatConversions {
+module {:options "-functionSyntax:4"} Uint8_16  refines LittleEndianNatConversions {
```
Comment (davidcok) [inline src/Collections/Sequences/LittleEndianNatConversions.dfy:225]:
```suggestion
module {:options "-functionSyntax:4"} Uint8_16 refines LittleEndianNatConversions {
```

File: src/Collections/Sequences/LittleEndianNatConversions.dfy [hunk 2]
```diff
@@ -239,15 +239,15 @@ module Uint8_16 refines LittleEndianNatConversions {
 }
 
 /* Conversions between sequences of uint8 and uint32. */
-module Uint8_32 refines LittleEndianNatConversions {
+module {:options "-functionSyntax:4"} Uint8_32  refines LittleEndianNatConversions {
```
Comment (davidcok) [inline src/Collections/Sequences/LittleEndianNatConversions.dfy:242]:
```suggestion
module {:options "-functionSyntax:4"} Uint8_32 refines LittleEndianNatConversions {
```

File: src/Collections/Sequences/LittleEndianNatConversions.dfy [hunk 3]
```diff
@@ -256,15 +256,15 @@ module Uint8_32 refines LittleEndianNatConversions {
 }
 
 /* Conversions between sequences of uint8 and uint64. */
-module Uint8_64 refines LittleEndianNatConversions {
+module {:options "-functionSyntax:4"} Uint8_64  refines LittleEndianNatConversions {
```
Comment (davidcok) [inline src/Collections/Sequences/LittleEndianNatConversions.dfy:259]:
```suggestion
module {:options "-functionSyntax:4"} Uint8_64 refines LittleEndianNatConversions {
```

File: src/Collections/Sequences/LittleEndianNatConversions.dfy [hunk 4]
```diff
@@ -273,15 +273,15 @@ module Uint8_64 refines LittleEndianNatConversions {
 }
 
 /* Conversions between sequences of uint16 and uint32. */
-module Uint16_32 refines LittleEndianNatConversions {
+module {:options "-functionSyntax:4"} Uint16_32  refines LittleEndianNatConversions {
```
Comment (davidcok) [inline src/Collections/Sequences/LittleEndianNatConversions.dfy:276]:
```suggestion
module {:options "-functionSyntax:4"} Uint16_32 refines LittleEndianNatConversions {
```

File: src/Collections/Sequences/LittleEndianNatConversions.dfy [hunk 5]
```diff
@@ -290,15 +290,15 @@ module Uint16_32 refines LittleEndianNatConversions {
 }
 
 /* Conversions between sequences of uint32 and uint64. */
-module Uint32_64 refines LittleEndianNatConversions {
+module {:options "-functionSyntax:4"} Uint32_64  refines LittleEndianNatConversions {
```
Comment (davidcok) [inline src/Collections/Sequences/LittleEndianNatConversions.dfy:293]:
```suggestion
module {:options "-functionSyntax:4"} Uint32_64 refines LittleEndianNatConversions {
```

File: src/Unicode/Utf8EncodingForm.dfy [hunk 6]
```diff
@@ -15,14 +15,14 @@ include "UnicodeEncodingForm.dfy"
  * The Unicode encoding form that assigns each Unicode scalar value to an unsigned byte sequence of one to four bytes
  * in length, as specified in Table 3-6 and Table 3-7.
  */
-module Utf8EncodingForm refines UnicodeEncodingForm {
+module {:options "-functionSyntax:4"} Utf8EncodingForm  refines UnicodeEncodingForm {
```
Comment (davidcok) [inline src/Unicode/Utf8EncodingForm.dfy:18]:
```suggestion
module {:options "-functionSyntax:4"} Utf8EncodingForm refines UnicodeEncodingForm {
```

**Review by davidcok — APPROVED** [review]
Didn't realize you were going through the whole library -- but just as well.
Just suggested removing some extraneous white space

**Review by atomb — APPROVED** [review]
This looks good to me (though good catch on the extra spaces @davidcok).

**Review by atomb — APPROVED** [review]

**Pull request merged**